level1a/handlers/level1a.py

```diff
@@ -165,28 +165,46 @@ def interp_array(
     if np.min(diffs) == 0:
         return values[np.argmin(diffs)]
     dt = np.diff(indices)
-    weights = diffs[::-1]/dt
+    weights = diffs[::-1] / dt
     return np.average(values, weights=weights, axis=0)
 
 
 def interp_to(
     target_date: Timestamp,
     column: Series,
+    max_diff: Optional[Timedelta] = None,
 ) -> np.ndarray:
     before = column.index.get_indexer([target_date], method='ffill')[0]
     after = column.index.get_indexer([target_date], method='bfill')[0]
 
+    if (before < 0 or after < 0) or (before > after):
+        return column.iloc[0] * np.nan
+
     timestamps = column.index[[before, after]].astype(int).values
+
+    if (
+        max_diff is not None
+        and (
+            Timedelta(seconds=np.abs(timestamps[1] - timestamps[0]) * 1e-9)
+            > max_diff
+        )
+    ):
+        return column.iloc[0] * np.nan
+
     return interp_array(
         target_date.value,
         timestamps,
         column.iloc[[before, after]].values,
     )
```

Comment:
```suggestion
    timestamps = column.index[[before, after]]

    if (
        max_diff is not None
        and np.diff(timestamps)[0] > max_diff
    ):
        return column.iloc[0] * np.nan

    return interp_array(
        target_date.value,
        timestamps.astype(int).values,
        column.iloc[[before, after]].values,
    )
```
I had a hard time understanding what the `if`-statement does, so maybe something like this instead?